provider/google/metadata_test.go

```diff
@@ -0,0 +1,98 @@
+package google
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/taskcluster/httpbackoff"
+)
+
+type fakeMetadataService struct {
+	UserDataError error
+	UserData      *UserData
+	Metadata      map[string]string
+}
+
+func (mds *fakeMetadataService) queryUserData() (*UserData, error) {
+	if mds.UserDataError != nil {
+		return nil, mds.UserDataError
+	}
+	return mds.UserData, nil
+}
+
+func (mds *fakeMetadataService) queryMetadata(path string) (string, error) {
+	if path[0] != '/' {
+		panic("path must start with /")
+	}
+	res, ok := mds.Metadata[path]
+	if !ok {
+		return "", fmt.Errorf("not found: %s", path)
+	}
+	return res, nil
+}
+
+func TestQueryMetadata(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Header.Get("Metadata-Flavor") != "Google" {
+			w.WriteHeader(400)
+			fmt.Fprintln(w, "Metadata-Flavor Missing")
+		} else if r.URL.Path == "/computeMetadata/v1/project/id" {
+			w.WriteHeader(200)
+			fmt.Fprintln(w, "42")
+		} else {
+			w.WriteHeader(404)
+			fmt.Fprintln(w, "Not Found")
+		}
+	}))
+	defer ts.Close()
+
+	metadataBaseURL = ts.URL + "/computeMetadata/v1/project"
```

Comment:
Why not create a new string here instead?

Reply:
I'm not sure what you mean -- use `fmt.Sprintf`?

Reply:
Never mind - this makes sense, I was misinterpreting what you were doing here. You need to use `metadataBaseURL`, it's what is used in the package.